nano/node/blockprocessor.hpp

```diff
@@ -56,7 +56,7 @@ class block_processor final
 
 public: // Events
 	using processed_t = std::pair<nano::process_return, std::shared_ptr<nano::block>>;
-	nano::observer_set<nano::process_return const &, std::shared_ptr<nano::block>> processed;
+	nano::observer_set<nano::process_return const &, std::shared_ptr<nano::block>, nano::transaction const &> processed;
```

Comment:
I'd rather we didn't put transactions through this callback. Rather I think this should use the batch_processed observer and within that observer, create the required transaction so the observers themselves don't need to be modified.